tools/autograph-monitor/lambda-setup-entrypoint.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+set -e
+set -o pipefail
+
+# Duplicate the console output to a file.
+exec &> >(tee /tmp/autograph-lambda-logs.txt)
+
+# Wait for the heartbeat
+HEARTBEAT=$(curl --silent \
+                 --connect-timeout 5 \
+                 --max-time 10 \
+                 --retry-connrefused \
+                 --retry 5 \
+                 --retry-delay 5 \
+                 --retry-max-time 60 \
+                 "${AUTOGRAPH_URL}/__heartbeat__")
+RETCODE=$?
+if [ $RETCODE -ne 0 ]; then
+    echo "Failed to reach autograph heartbeat" >&2
+    exit $RETCODE
+fi
+
+# Fetch the normandy root hash
+export AUTOGRAPH_ROOT_HASH=$(autograph-client -t "$AUTOGRAPH_URL" -listconfig normandy | \
```

Comment:
So, I've been thinking about this over the weekend. You've put a lot of work into this and it's good, but I'm kind of not sure about the goal here. My understanding of this ROOT_HASH variable was to prevent configurations from accidentally being changed when we didn't want them to be changed. This seems to make this check not actually a backstop check anymore. Are we sure we want to do this?

Reply:
(If we wanted to make sure we were future proofed for changes to existing signer configs' chains, I'd recommend adding key agility here, instead, and not doing this config lookup work.)

Reply:
It's all a bit of a moot point as I am not sure if the monitor lambda was ever deployed anywhere, but this doesn't actually change anything in how the monitor lambda works, and the `lambda-setup-entrypoint.sh` script here is only being used in the test containers. In theory, if a lambda was deployed in production, it's value of `AUTOGRAPH_ROOT_HASH` would be hardcoded in the AWS environment and it would serve as the backstop check as intended.

Reply:
That being said, I do admit that I very much over-engineered the solution :)

Reply:
Okay, so, in the time since this and while you were out, Jon and I have figured out that the monitor lambda _does_ actually run, and _is_ actually erroring. Interestingly, those errors are different in stage and in prod. I think we could drop back this change to the non-config stuff, update the normandy root key value in autograph.softhsm.yaml and weave it through the other bits, and still get some success. Do you think so? 

Maybe I'm missing a piece, though.

Reply:
Okay - I have rebased this work to abandon the config API, and implemented the monitor test for the non-HSM version only. I don't know if the monitor-hsm test really adds any value right now anyways.